plugins/woocommerce-blocks/assets/js/blocks/product-collection/frontend.tsx

```diff
@@ -164,6 +167,12 @@ const productCollectionStore = {
 				yield prefetch( ref.href );
 			}
 		},
+		*viewProduct() {
+			const context = getContext< Context >();
```

Comment:
Looks like there is a TS error
```suggestion
			const context = getContext< {
				productId: number;
			} >();
```

Reply:
Good catch, I extended `ProductCollectionStoreContext` with `productId` and improved types 👍